src/Features/Core/Portable/PullMemberUp/PullMemberUpAnalyzer.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.  
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp
+{
+    internal class PullMembersUpAnalysisBuilder
+    {
+        private static bool IsAbstractModifiersMatch(INamedTypeSymbol targetSymbol, ISymbol selectedMember)
+        {
+            return !selectedMember.IsAbstract || targetSymbol.IsAbstract;
+        }
+
+        private static bool IsAccessiblityModifiersMatch(INamedTypeSymbol targetSymbol, ISymbol selectedMember)
+        {
+            return selectedMember.DeclaredAccessibility == Accessibility.Public;
+        }
+
+        private static bool IsStaticModifiersMatch(INamedTypeSymbol targetSymbol, ISymbol selectedMember)
+        {
+            return !selectedMember.IsStatic;
```

Comment:
1. this impl doesn't match the title of the method.
 2. you don't use targetsymbol.